src/mqt/problemsolver/csp.py

```diff
@@ -202,7 +231,22 @@ def create_oracle(
 
         return qc.to_instruction(label="oracle")
 
-    def init_qc(self):
+    def init_qc(
+        self,
+    ) -> tuple[
+        QuantumCircuit,
+        QuantumRegister,
+        QuantumRegister,
+        QuantumRegister,
+        int,
+        int,
+        tuple[
+            tuple[QuantumRegister, QuantumRegister],
+            tuple[QuantumRegister, QuantumRegister],
+            tuple[QuantumRegister, QuantumRegister],
+            tuple[QuantumRegister, QuantumRegister],
+        ],
+    ]:
```

Comment:
Oh wow, what a monstrosity of a return type 😅
I am fine with it at the moment, but this is also something to be improved in the future.